tchannel/tornado/connection.py

```diff
@@ -701,3 +633,162 @@ def _request_timed_out(self, req_id, req_ttl, future):
         # while.
         future.set_exception(errors.TimeoutError())
         self._request_tombstones.add(req_id, req_ttl)
+
+
+class Reader(object):
+
+    def __init__(self, io_stream, capacity=64):
+        self.queue = tornado.queues.Queue(capacity)
+        self.filling = False
+        self.io_stream = io_stream
+
+    def fill(self):
+        self.filling = True
+
+        io_loop = IOLoop.current()
+
+        def keep_reading(f):
+            if f.exception():
+                return log(f.exception())
```

Comment:
You want to spawn a callback even in case of an exception, don't you? Or do you want it to stop filling once an error happens. Also, you should set filling to False if you stop.